src/cli/constant.js

```diff
@@ -234,6 +234,7 @@ const detailedOptions = normalizeDetailedOptions({
       "less",
       "scss",
       "json",
+      // "glimmer",
```

Comment:
I would name all those "handlebars". Except for JavaScript, we started naming the parser after the language it parses and not the actual name of the parser. This is easier for people to figure out which to use. It would be nice to rename it to --language at some point but we'll see.

Reply:
I agree that renaming from glimmer to handlebars would make sense. It's just that at this moment, glimmer doesn't support partial statements, which handlebars does support. So it's kind of a specific caveat that they're different.

Reply:
Do you know if there's a list of the differences between the two?

Reply:
I'm not sure, I'll ask the maintainers of the project.

Reply:
Handlebars is a plain text templating language. In practice it is often used as an HTML templating language, but it is also commonly used for templating configuration files (toml, ini, yaml, etc).

Glimmer is an HTML templating language, but there are a few differences from vanilla HTML:
- The parser will hard error if non-void tags aren't balanced. I don't think we plan to change that any time soon.
- There are a few extensions we've added to HTML that let us differentiate HTML attributes from component arguments:

```hbs
<MyComponent class="red" @data={{myData}}>
```

You can also pass blocks as arguments (I believe they are commonly called render props in React)

```hbs
<MyComponent>
  <@header=>This is a header!</@header>
  <@header=>This is a footer!</@header>
</MyComponent>
```

Reply:
@mmun isn't it technically "HTMLBars" instead of "Glimmer" with "Glimmer" just implementing a parser for "HTMLBars"?

Reply:
It's true that that is [how we launched it](https://emberjs.com/blog/2015/02/07/ember-1-10-0-released.html) but it was always for lack of a better name. You won't find references to HTMLBars in the guides/API docs anymore. The Glimmer brand now encompasses a DOM templating language, a runtime VM, and an experimental application framework (http://glimmerjs.com)

Reply:
@vjeux Let me start by saying thank you very much for working on prettier and for reviewing this PR! And thanks to @robert-j-webb for working on it!

I completely agree with you that we should minimize confusion with the configuration. Taking a step back away from Glimmer for a second, what configuration would you propose to differentiate between `.txt.hbs`, `.html.hbs`, `.toml.hbs`, etc? Each variant would need different formatting for the non-Handlebars parts. It's also plausible to me that the stuff inside the curlies (`{{ ... }}`) might be slightly formatted differently depending on the variant as well.

Reply:
I’m not super familiar with handlebars but the philosophy we followed with prettier so far is to take use cases one by one and make them really good. So my recommendation would be to make it work perfectly for Ember and then we can figure out how to handle more use cases. My guy feeling is that there’s a big market for people eager to get prettier to work for the html variant and low interest in the other variants right now.